lib/board.rb

```diff
@@ -0,0 +1,69 @@
+class Board
+  attr_accessor :cells
+
+  def reset!
+    # can reset the state of the cells in the board
+    # sets the cells of the board to a 9 element array of " "
+    @cells = Array.new(9, " ")
+  end
+  def initialize
+    # sets the cells of the board to a 9 element array of " "
+    @cells = Array.new(9, " ")
+  end
+  def display
+    # prints the board
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+  def position(index)
+    # takes in user input and returns the value of the board cell
+    idx = input_to_index(index)
+    @cells[idx]
+  end
+  def full?
+    # returns true for a full board
+    # returns false for an in-progress game
+    @cells.all? {|token| token == "X" || token == "O"}
+  end
+  def turn_count
+    # returns the amount of turns based on cell value
+    count = 0
+    @cells.each do |play|
+      if play == "O" || play == "X"
+        count +=1
+      end
+    end
+    count
+  end
+  def taken?(index)
+    # returns true if the position is X or O
+    # returns false if the position is empty or blank
+    idx = input_to_index(index)
+    if @cells[idx] == "O" || @cells[idx] == "X"
+      true
+    else
+      false
+    end
+  end
```

Comment:
Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression.  This applies to other methods as well, including `#valid_move?`, `Game#over?`, and `Game#draw?`.